src/commands/run.rs

```diff
@@ -425,7 +430,29 @@ fn populate_with_wasi(
         for (name, dir) in preopen_dirs.into_iter() {
             builder = builder.preopened_dir(dir, name)?;
         }
-        store.data_mut().wasi = Some(builder.build());
+
+        let mut ctx = builder.build();
+
+        let mut listenfd = ListenFd::from_env();
```

Comment:
I think this shouldn't be done by default. I think it should only give access if a commandline argument is passed to enable it. In addition it would be nice to have a commandline argument to manually specify ports to listen to without having to use systemd or some other external program to create the socket.

Reply:
See the `TODO` section in the PR comment :)

Reply:
Happy to receive PRs to this draft :) Will add you as a co-author

Reply:
Right. I honestly only looked at the source and barely at the description :sweat_smile:

Reply:
moved CLI PR to https://github.com/bytecodealliance/wasmtime/pull/3729